### ISSUE TITLE
🐛 java.lang.UnsatisfiedLinkError: dlopen failed: library "libllm_inference_engine_jni.so" not found

### ISSUE THREAD

**Issue by tempo-riz** [DenisovAV/flutter_gemma#9]
I downloaded then copied the model gemma-2b-it-cpu-int4.bin that i got from kaggle, this one 
![image](https://github.com/DenisovAV/flutter_gemma/assets/64015162/21b4d5ce-a52d-4f9e-9a49-1f62ebf5f1bc)
(is it the correct one  @DenisovAV ?)

then called `Gemma.instance.init(
      maxTokens: 512,
    );`

on a physical device : 
```
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: PlatformException(ERROR, Failed to initialize gemma, Field modelPath_ for n6.b not found. Known fields are [private int n6.b.j, private java.lang.String n6.b.k, private java.lang.String n6.b.l, private int n6.b.m, private int n6.b.n, private int n6.b.o, private float n6.b.p, private int n6.b.q, private static final n6.b n6.b.r, private static volatile q6.a1 n6.b.s], null)
```

in the emulator :
```
E/AndroidRuntime(30074): java.lang.UnsatisfiedLinkError: dlopen failed: library "libllm_inference_engine_jni.so" not found
E/AndroidRuntime(30074): 	at java.lang.Runtime.loadLibrary0(Runtime.java:1082)
E/AndroidRuntime(30074): 	at java.lang.Runtime.loadLibrary0(Runtime.java:1003)
E/AndroidRuntime(30074): 	at java.lang.System.loadLibrary(System.java:1661)
E/AndroidRuntime(30074): 	at p6.b.<clinit>(Unknown Source:10)
E/AndroidRuntime(30074): 	at u6.c.<init>(Unknown Source:53)
E/AndroidRuntime(30074): 	at u6.c.<init>(Unknown Source:0)
E/AndroidRuntime(30074): 	at u6.c$a.a(Unknown Source:22)
E/AndroidRuntime(30074): 	at u6.a.onMethodCall(Unknown Source:51)
E/AndroidRuntime(30074): 	at o7.j$a.a(Unknown Source:17)
E/AndroidRuntime(30074): 	at c7.c.l(Unknown Source:18)
E/AndroidRuntime(30074): 	at c7.c.m(Unknown Source:41)
E/AndroidRuntime(30074): 	at c7.c.i(Unknown Source:0)
E/AndroidRuntime(30074): 	at c7.b.run(Unknown Source:12)
E/AndroidRuntime(30074): 	at android.os.Handler.handleCallback(Handler.java:958)
E/AndroidRuntime(30074): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/AndroidRuntime(30074): 	at android.os.Looper.loopOnce(Looper.java:205)
E/AndroidRuntime(30074): 	at android.os.Looper.loop(Looper.java:294)
E/AndroidRuntime(30074): 	at android.app.ActivityThread.main(ActivityThread.java:8177)
E/AndroidRuntime(30074): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(30074): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
E/AndroidRuntime(30074): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
```

**Comment by studentsdav** [comment]
same issue

E/AndroidRuntime(11679): java.lang.UnsatisfiedLinkError: dlopen failed: library "libllm_inference_engine_jni.so" not found
E/AndroidRuntime(11679): 	at java.lang.Runtime.loadLibrary0(Runtime.java:1082)
E/AndroidRuntime(11679): 	at java.lang.Runtime.loadLibrary0(Runtime.java:1003)
E/AndroidRuntime(11679): 	at java.lang.System.loadLibrary(System.java:1661)
E/AndroidRuntime(11679): 	at com.google.mediapipe.tasks.genai.llminference.LlmInference.<clinit>(LlmInference.java:27)
E/AndroidRuntime(11679): 	at dev.flutterberlin.flutter_gemma.InferenceModel.<init>(InferenceModel.kt:41)
E/AndroidRuntime(11679): 	at dev.flutterberlin.flutter_gemma.InferenceModel.<init>(Unknown Source:0)
E/AndroidRuntime(11679): 	at dev.flutterberlin.flutter_gemma.InferenceModel$Companion.getInstance(InferenceModel.kt:60)
E/AndroidRuntime(11679): 	at dev.flutterberlin.flutter_gemma.FlutterGemmaPlugin.onMethodCall(FlutterGemmaPlugin.kt:42)
E/AndroidRuntime(11679): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:267)
E/AndroidRuntime(11679): 	at io.flutter.embedding.engine.dart.DartMessenger.invokeHandler(DartMessenger.java:292)
E/AndroidRuntime(11679): 	at io.flutter.embedding.engine.dart.DartMessenger.lambda$dispatchMessageToQueue$0$io-flutter-embedding-engine-dart-DartMessenger(DartMessenger.java:319)
E/AndroidRuntime(11679): 	at io.flutter.embedding.engine.dart.DartMessenger$$ExternalSyntheticLambda0.run(Unknown Source:12)
E/AndroidRuntime(11679): 	at android.os.Handler.handleCallback(Handler.java:958)
E/AndroidRuntime(11679): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/AndroidRuntime(11679): 	at android.os.Looper.loopOnce(Looper.java:205)
E/AndroidRuntime(11679): 	at android.os.Looper.loop(Looper.java:294)
E/AndroidRuntime(11679): 	at android.app.ActivityThread.main(ActivityThread.java:8177)
E/AndroidRuntime(11679): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(11679): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
E/AndroidRuntime(11679): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
I/Process (11679): Sending signal. PID: 11679 SIG: 9
Lost connection to device.

**Comment by DenisovAV** [comment]
the rootcause of issue is a pro-guard obfuscation, it should be reproduced only in release mode, I fixed it in version 0.2.2 please check
